docs: Proposed cleanup and simplification (#663)
Consensys/doc.linea: docs/architecture/index.mdx

## Linea's ideal state

> Linea's aims to be a fully decentralized, permissionless network. To support this goal, the Linea architecture is made up of three main elements:

Reviewer: ```suggestion
Linea aims to be a fully decentralized, permissionless network. To support this goal, the Linea architecture is made up of three main elements:
```

## First of all: what _is_ Linea, anyway? What's a zkEVM L2?

There are several different mental models that people use to explain the different
networks and how they relate to one another. Some people prefer to call L2s "rollup networks", 
or "scaling solutions". The term "layer" is a popular mental model: where one network is "built on top of another". 

Let's set metaphor aside for a moment, and clarify:

**The Ethereum network has several functional areas:**

- _Blockchain_: keeps track of addresses, and which tokens are allocated to 
which addresses
- _Consensus_ mechanism: wherein many many nodes communicate about the movement of 
tokens from one address to another, and each keeps their local copy of the ledger up to date
- _Execution environment_: the EVM (Ethereum Virtual Machine) wherein computer programs can be run. 

These three areas are heavily interconnected, and this is a simplification, but it's a helpful 

> one to understand what's going on when adding a layer.

Reviewer: ```suggestion
one to understand the tradeoffs all networks must make.
```

Ethereum prioritizes security: that consensus mechanism is designed to ensure that no one can 
singlehandedly alter the state of the network. This is a very impressive feat of engineering, 
and it comes with a tradeoff: the execution environment is highly limited in the amount of work 
it can do, because the consensus mechanism intentionally runs slowly, to keep everything safe.

> Linea, and other L2s are designed to participate in the security of Ethereum, 

Reviewer: ```suggestion
Linea and other L2s are designed to participate in the security mechanism of Ethereum, 
```

> while optimizing for execution. In other words, L2s support many transactions, run 

Reviewer: ```suggestion
while optimizing for execution. In other words, L2s enables users to make many transactions, run 
```

Reviewer: Not sure the rest of the sentence works without inserting users as the subject

  </div>
</div>

## Next steps

> - Learn more about the different components that make up the [Linea software](architecture/stack).

Reviewer: ```suggestion
- Learn more about the different components that comprise [Linea](architecture/stack).
```

- Understand the first point of contact for many new L2 users, the bridge. Linea has more than one bridge; and that number is likely to continue to 

> grow. Key these bridges and how they function is the [Linea canonical message service](architecture/stack/canonical-msg-service).

Reviewer: ```suggestion
grow. The [Linea canonical message service](architecture/stack/canonical-msg-service) is key to these bridges and how they function.
```
